hackathon/views.py

```diff
@@ -1,10 +1,61 @@
 from django.views.generic import ListView
+from django.shortcuts import render, get_object_or_404, redirect
+from django.contrib import messages
+from django.contrib.auth.decorators import login_required
+from datetime import datetime
 
-from .models import Hackathon
+from .models import Hackathon, HackAwardCategory, HackTeam, HackProject, HackProjectScore, HackProjectScoreCategory
+from .forms import HackathonForm
 
 
 class HackathonListView(ListView):
     """Renders a page with a list of Hackathons."""
     model = Hackathon
     ordering = ['-created']
     paginate_by = 8
+
+
+@login_required
+def create_event(request):
+    """ Allow users to create hackathon event """
+
+    if request.method == 'GET':
+        # Redirect user if they are not admin
+        if not request.user.is_superuser:
+            return redirect("hackathon:hackathon-list")
+
+        template = "hackathon/create-event.html"
+        form = HackathonForm()
+        context = {
```

Comment:
Not sure you need to create a context dict just for one key. I'd just pass the form dict directly to the `render` function.

Reply:
This is just a personal preference, but changed as requested.